backend/application/syson-application/src/test/java/org/eclipse/syson/application/controllers/diagrams/general/view/GVAddExistingElementsTests.java

```diff
@@ -179,78 +182,114 @@ public void addExistingElementsRecursiveOnDiagram() {
-                            .isPresent();
-                    assertThat(action2ActionFlowCompartment.get().getChildNodes())
-                            .as(ACTION2 + " action flow compartment should contain 1 child")
-                            .hasSize(1)
-                            .as(ACTION2 + " action flow compartment should contain " + ACTION3)
-                            .anyMatch(n -> Objects.equals(n.getTargetObjectLabel(), ACTION3));
+                    checkAction2(action1ActionFlowCompartment);
+
+                    checkRequirementUsage(newDiagram);
 
                 }, () -> fail("Missing diagram"));
         this.verifier.consumeNextWith(updatedDiagramConsumer);
     }
 
+    private static void checkPackageNode(Diagram newDiagram) {
+        var optPackageNode = newDiagram.getNodes().stream()
+                .filter(n -> Objects.equals(n.getTargetObjectLabel(), PACKAGE1))
+                .findFirst();
+        assertThat(optPackageNode).isPresent();
+        assertThat(optPackageNode.get().getChildNodes())
+                .as("Node " + PACKAGE1 + " should contain 1 child")
+                .hasSize(1)
+                .as("Node " + ATTRIBUTE_DEFINITION + " should exist inside " + PACKAGE1)
+                .anyMatch(n -> Objects.equals(n.getTargetObjectLabel(), ATTRIBUTE_DEFINITION));
+    }
+
+    private void checkAction2(Optional<Node> action1ActionFlowCompartment) {
+        var optAction2Node = action1ActionFlowCompartment.get().getChildNodes().stream()
+                .filter(n -> Objects.equals(n.getTargetObjectLabel(), ACTION2))
+                .findFirst();
+        assertThat(optAction2Node).isPresent();
+        var action2Node = optAction2Node.get();
+        var action2ActionsCompartment = this.getCompartment(action2Node, "actions");
+        assertThat(action2ActionsCompartment)
+                .as(ACTION2 + " should contain an actions compartment")
+                .isPresent();
+        assertThat(action2ActionsCompartment.get().getChildNodes())
+                .as(ACTION2 + " actions compartment should contain 1 child")
+                .hasSize(1)
+                .as(ACTION2 + " actions compartment should contain " + ACTION3)
+                .anyMatch(n -> Objects.equals(n.getTargetObjectLabel(), ACTION3));
+
+        var action2ActionFlowCompartment = this.getCompartment(action2Node, "action flow");
+        assertThat(action2ActionFlowCompartment)
+                .as(ACTION2 + " should contain an action flow compartment")
+                .isPresent();
+        assertThat(action2ActionFlowCompartment.get().getChildNodes())
+                .as(ACTION2 + " action flow compartment should contain 1 child")
+                .hasSize(1)
+                .as(ACTION2 + " action flow compartment should contain " + ACTION3)
+                .anyMatch(n -> Objects.equals(n.getTargetObjectLabel(), ACTION3));
+    }
+
+    private @NotNull Optional<Node> checkAction1(Diagram newDiagram) {
```

Comment:
You could also return a Node and a check about the nullity at the end of the method